kreuzberg/_chunker.py

```diff
@@ -36,7 +38,7 @@ def get_chunker(
     if key not in _chunkers:
         try:
             if mime_type == MARKDOWN_MIME_TYPE:
-                from semantic_text_splitter import MarkdownSplitter
+                from semantic_text_splitter import MarkdownSplitter  # type: ignore[import-not-found]
```

Comment:
ok, the issue you had and why you needed t do all this changes is that you did not install the extras.

Look under pyproject.toml to understand how optional dependencies work. Read in the uv sync docs about how to install all extras. 

Most of the typing changes you did are unnecessary, you simply didnt have the extras in place.